src/sinks/elasticsearch.rs

```diff
@@ -94,6 +95,35 @@ impl ElasticSearchAuth {
     }
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub enum BulkAction {
+    Index,
+    Create,
+}
+
+impl Default for BulkAction {
+    fn default() -> Self {
+        BulkAction::Index
+    }
+}
+
+impl BulkAction {
+    pub fn as_str(&self) -> &'static str {
+        match *self {
+            BulkAction::Index => "index",
+            BulkAction::Create => "create",
+        }
+    }
+
+    pub fn as_path(&self) -> &'static str {
```

Comment:
Maybe `as_json_pointer`. I initially thought this was going to be used as an HTTP path.